featurebyte/models/entity_lookup_feature_table.py

```diff
@@ -0,0 +1,240 @@
+                )
+            ],
+            source_type=feature_store.type,
+        )
+        entity_universe = EntityUniverseModel(
+            query_template=SqlglotExpressionModel.create(universe_expr)
+        )
+        entity_lookup_feature_table_model = OfflineStoreFeatureTableModel(
+            name=get_lookup_feature_table_name(lookup_step.id),
+            feature_ids=[],
+            primary_entity_ids=[lookup_step.child_entity.id],
+            serving_names=[lookup_step.child_entity.serving_names[0]],
+            feature_cluster=feature_cluster,
+            output_column_names=[lookup_step.parent_entity.serving_names[0]],
+            output_dtypes=[lookup_graph_result.feature_dtype],
+            entity_universe=entity_universe,
+            has_ttl=False,
+            feature_job_setting=FeatureJobSetting(
+                frequency="1d",
+                time_modulo_frequency="0s",
+                blind_spot="0s",
+            ),
+            is_entity_lookup=True,
+        )
+        out.append(entity_lookup_feature_table_model)
+    return out
+
+
+def _get_entity_lookup_graph(
+    lookup_step: EntityLookupStep,
+    feature_store: FeatureStoreModel,
+) -> EntityLookupGraphResult:
+    relation_table = lookup_step.relation_table
+    graph = QueryGraph()
+    input_node = graph.add_operation_node(
+        node=relation_table.construct_input_node(feature_store_details=feature_store),
+        input_nodes=[],
+    )
+
+    child_column_name = None
+    parent_column_name = None
+    feature_dtype = None
+    for column_info in relation_table.columns_info:
+        if column_info.entity_id == lookup_step.child_entity.id:
+            child_column_name = column_info.name
+        elif column_info.entity_id == lookup_step.parent_entity.id:
+            parent_column_name = column_info.name
+            feature_dtype = column_info.dtype
+    assert child_column_name is not None
+    assert parent_column_name is not None
+    assert feature_dtype is not None
+
+    additional_params: Dict[str, Any]
+    if relation_table.type == TableDataType.SCD_TABLE:
+        assert isinstance(relation_table, SCDTableModel)
+        additional_params = {
+            "scd_parameters": SCDBaseParameters(
+                effective_timestamp_column=relation_table.effective_timestamp_column,
+                natural_key_column=relation_table.natural_key_column,
+                current_flag_column=relation_table.current_flag_column,
+                end_timestamp_column=relation_table.end_timestamp_column,
+            )
+        }
+    elif relation_table.type == TableDataType.EVENT_TABLE:
+        assert isinstance(relation_table, EventTableModel)
+        additional_params = {
+            "event_parameters": {
+                "event_timestamp_column": relation_table.event_timestamp_column,
+            }
+        }
+    else:
+        # TODO: handle ITEM_TABLE which also needs event_parameters
+        additional_params = {}
```

Comment:
To handle this, thinking of add a special handling for table serving like the following:
```
table = await table_service.get_document(document_id=table_id, populate_reference_table=True)
# if populate_reference_table is True, it will also retrieve the reference table (for item table, reference table is event table)
```
By doing so, handling item table should be easier here:
```
event_table_params = item_table.event_table.parameters
```

Reply:
That makes sense to me.